### ISSUE TITLE
[firebase_messaging]: Notifications received through APNS Cloud Kit but not Firebase

### ISSUE THREAD

**Issue by ahumeniy** [firebase/flutterfire#11155]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Are you aware of the differences between iOS and Android background message handling?

- [X] I understand that iOS and Android background messages behave differently, and I've designed my application with that in mind.

### Do you have an active Apple Developer account?

- [X] I have an active Apple Developer account.

### Are you using a physical iOS device to test background messages?

- [X] I am using a physical iOS device to test background messages.

### Have you enabled "Remote Notifications" & "Background Mode" (Checking options for "Background Processing" & "Remote Notifications") in your app's Xcode project?

![Screenshot 2023-06-19 at 14 31 38](https://github.com/firebase/flutterfire/assets/6594630/cdab7afb-68b6-49ff-9c38-b3de0477988f)

### Have you created an APNs key in your Apple Developer account & uploaded this APNs key to your Firebase console?

![Screenshot 2023-06-19 at 14 34 52](https://github.com/firebase/flutterfire/assets/6594630/e428b172-792d-4d41-8c44-86e80afa76ad)

### Have you disabled method swizzling for Firebase in your app?

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>CADisableMinimumFrameDurationOnPhone</key>
    <true/>
    <key>CFBundleDevelopmentRegion</key>
    <string>$(DEVELOPMENT_LANGUAGE)</string>
    <key>CFBundleDisplayName</key>
    <string>Polpaico Clone</string>
    <key>CFBundleDocumentTypes</key>
    <array>
      <string>com.apple.maps.directionsrequest</string>
    </array>
    <key>CFBundleExecutable</key>
    <string>$(EXECUTABLE_NAME)</string>
    <key>CFBundleIdentifier</key>
    <string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
    <key>CFBundleInfoDictionaryVersion</key>
    <string>6.0</string>
    <key>CFBundleName</key>
    <string>polpaico_clone</string>
    <key>CFBundlePackageType</key>
    <string>APPL</string>
    <key>CFBundleShortVersionString</key>
    <string>$(FLUTTER_BUILD_NAME)</string>
    <key>CFBundleSignature</key>
    <string>????</string>
    <key>CFBundleURLSchemes</key>
    <array>
      <string>fb871423333934677</string>
      <string>maestropolpaico</string>
    </array>
    <key>CFBundleVersion</key>
    <string>$(FLUTTER_BUILD_NUMBER)</string>
    <key>FacebookAppId</key>
    <string>871423333934677</string>
    <key>FacebookDisplayName</key>
    <string>Maestro Polpaico</string>
    <key>LSRequiresIPhoneOS</key>
    <true/>
    <key>NSLocationTemporaryUsageDescriptionDictionary</key>
    <array>
      <dict>
        <key>TemporaryFullAccuracyUsageDescription</key>
        <string>Si permites que Locales use tu ubicación podrás saber qué sucursales de Polpaico están cerca de ti.</string>
      </dict>
    </array>
    <key>NSLocationWhenInUseUsageDescription</key>
    <string>Si permites que esta App use tu ubicación podrás saber qué sucursales de Polpaico están cerca de ti.</string>
    <key>UIApplicationSupportsIndirectInputEvents</key>
    <true/>
    <key>UIBackgroundModes</key>
    <array>
      <string>processing</string>
      <string>remote-notification</string>
    </array>
    <key>UIDeviceFamily</key>
    <array>
      <integer>1</integer>
    </array>
    <key>UILaunchStoryboardName</key>
    <string>LaunchScreen</string>
    <key>UIMainStoryboardFile</key>
    <string>Main</string>
    <key>UIStatusBarHidden</key>
    <false/>
    <key>UISupportedInterfaceOrientations</key>
    <array>
      <string>UIInterfaceOrientationPortrait</string>
    </array>
    <key>UISupportedInterfaceOrientations~ipad</key>
    <array>
      <string>UIInterfaceOrientationPortrait</string>
      <string>UIInterfaceOrientationPortraitUpsideDown</string>
      <string>UIInterfaceOrientationLandscapeLeft</string>
      <string>UIInterfaceOrientationLandscapeRight</string>
    </array>
    <key>UIViewControllerBasedStatusBarAppearance</key>
    <false/>
  </dict>
</plist>
```

### Are you sending messages to your app from the Firebase Admin SDK?

I'm using the Firebase Console because that's what marketing people will use.

### Have you requested permission from the user to receive notifications?

- [X] I have the relevant permission to receive notifications.

### Have you used the 'Console' application on your macOS device to check if the iOS device's system is throttling your background messages?

```
default	14:41:27.026309-0400	Runner	[VERBOSE-2:FlutterDarwinContextMetalImpeller.mm(35)] Using the Impeller rendering backend.
default	14:41:27.066986-0400	Runner	flutter: The Dart VM service is listening on http://127.0.0.1:52118/zxtIonZS0jo=/
default	14:41:27.116063-0400	Runner	flutter: AppModule started!
default	14:41:27.116862-0400	Runner	flutter: -- FWelcomeModule INITIALIZED
default	14:41:27.120995-0400	Runner	flutter: App was not opened from a notification.
default	14:41:27.181619-0400	Runner	flutter: User granted permission
default	14:41:27.182235-0400	Runner	flutter: cbDxEBG0qUAGuKdxpaNSZK:APA91bExWl-xysvRJpnyYXiIAXb_CYIGng_-nFcIrvwAlJ45JCZ7ejMqQA7cCXfWN2dhohD4d7FyML0AMKUnDFP5XKZXNVo-ha1-eKWtSWmA2NJIe0Cz-MJDhwCYmnzErfL4-lZfc1ZF
default	14:41:27.183574-0400	Runner	10.9.0 - [FirebaseMessaging][I-FCM001000] FIRMessaging Remote Notifications proxy enabled, will swizzle remote notification receiver handlers. If you'd prefer to manually integrate Firebase Messaging, add "FirebaseAppDelegateProxyEnabled" to your Info.plist, and set it to NO. Follow the instructions at:
https://firebase.google.com/docs/cloud-messaging/ios/client#method_swizzling_in_firebase_messaging
to ensure proper integration.
default	14:42:47.323853-0400	Runner	[VERBOSE-2:FlutterDartVMServicePublisher.mm(152)] Could not register as server for FlutterDartVMServicePublisher. Check your network settings and relaunch the application.
default	14:42:48.589679-0400	Runner	flutter: -- FHomeModule INITIALIZED
default	14:42:48.589728-0400	Runner	flutter: -- FWelcomeModule DISPOSED
default	14:51:07.809417-0400	Runner	[VERBOSE-2:FlutterDarwinContextMetalImpeller.mm(35)] Using the Impeller rendering backend.
default	14:51:07.839901-0400	Runner	flutter: The Dart VM service is listening on http://127.0.0.1:52121/0sB4HBAqbww=/
default	14:51:07.915124-0400	Runner	flutter: App was not opened from a notification.
default	14:51:07.932495-0400	Runner	10.9.0 - [FirebaseMessaging][I-FCM001000] FIRMessaging Remote Notifications proxy enabled, will swizzle remote notification receiver handlers. If you'd prefer to manually integrate Firebase Messaging, add "FirebaseAppDelegateProxyEnabled" to your Info.plist, and set it to NO. Follow the instructions at:
https://firebase.google.com/docs/cloud-messaging/ios/client#method_swizzling_in_firebase_messaging
to ensure proper integration.
default	14:51:07.938374-0400	Runner	flutter: User granted permission
default	14:51:07.939687-0400	Runner	flutter: cbDxEBG0qUAGuKdxpaNSZK:APA91bExWl-xysvRJpnyYXiIAXb_CYIGng_-nFcIrvwAlJ45JCZ7ejMqQA7cCXfWN2dhohD4d7FyML0AMKUnDFP5XKZXNVo-ha1-eKWtSWmA2NJIe0Cz-MJDhwCYmnzErfL4-lZfc1ZF
default	14:51:12.196989-0400	Runner	flutter: -- FHomeModule INITIALIZED
default	14:51:12.197549-0400	Runner	flutter: -- FWelcomeModule DISPOSED

```

### Additional context and comments

Sending notifications via Apple Cloud Kit works fine.

**Comment by darshankawar** [comment]
Thanks for the report @ahumeniy 
Can you take a look at [this](https://github.com/firebase/flutterfire/issues/10082) and underlying comments and see if it helps in your case ?

**Comment by ahumeniy** [comment]
> Thanks for the report @ahumeniy Can you take a look at [this](https://github.com/firebase/flutterfire/issues/10082) and underlying comments and see if it helps in your case ?

Hi, I checked the linked issue and compared the test project (I was looking for an example for days. It should be more clear in the documentation where to look for an example) however the test project implements the AppDelegate in Objective C versus the one I have in Swift but I noticed it doesn't have anything related to Firebase unlike the answers I found somewhere else saying you should initialize and wait for messages there. The documentation is very confusing, also the Swift portion tells you to use the Flutterfire tool but I didn't had access to the Firebase Console and my coworker working with the Android part didn't need to use it so I had to guess how to implement it. The documentation should include the instructions to implement Firebase manually as an option.

However I still can't make it work. I send notifications and nothing comes up in the debugger console. I get a "Lost connection to device" every time I send a message so I guess something may be happening but I'm not sure if it's the messages because I regularly get "liquid detected in the connector" messages on this phone. The XCode console shows a lot of stuff but nothing sounding like a notification. What should I expect on it? Any ideas how can I filter it? Filtering just for "Runner" gives a lot of stuff, but if I filter by the package ID (com.example.something) there are more messages.

**Comment by ahumeniy** [comment]
I tried to run the `send-message` script from the example and at first it didn´t run saying my APNs SSL certificate was not valid or expired. I had an APNs Authentication Key set up, not SSL certificates so I deleted the APNs Key and put a certificate instead and it works now. Closing this, but Google should provide feedback on the Firebase console about errors while sending messages. Troubleshooting this kind of issues should be easier without the need for external tools.
